agent/job_runner.go

```diff
@@ -219,6 +219,7 @@ func NewJobRunner(l logger.Logger, scope *metrics.Scope, ag *api.AgentRegisterRe
 	runner.process = process.New(l, process.Config{
 		Path:            cmd[0],
 		Args:            cmd[1:],
+		Dir:             conf.AgentConfiguration.BuildPath,
```

Comment:
This is the main part of this PR - setting an explicit working directory for the bootstrap process. Everything else in the PR is a change that's required to make this line possible.